internal/controller/management/projectconfigs/project_configs.go

```diff
@@ -0,0 +1,343 @@
+package projectconfigs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"path"
+	"strings"
+	"time"
+
+	multierr "github.com/hashicorp/go-multierror"
+	"github.com/kelseyhightower/envconfig"
+	corev1 "k8s.io/api/core/v1"
+	kubeerr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/conditions"
+	"github.com/akuity/kargo/internal/controller"
+	"github.com/akuity/kargo/internal/kubeclient"
+	"github.com/akuity/kargo/internal/logging"
+	"github.com/akuity/kargo/internal/webhook/external"
+)
+
+type ReconcilerConfig struct {
+	Host                    string `envconfig:"HOST" default:"localhost"`
+	Port                    string `envconfig:"PORT" default:"8080"`
```

Comment:
These will be too easily mistaken for the controller's own host and port.

If we keep it consistent with how we've done this elsewhere, this should be:

```
ExternalWebhookServerBaseURL        string `envconfig:"EXTERNAL_WEBHOOK_SERVER_BASE_URL"`
```

A corresponding change will need to be made to set this env var in `charts/kargo/templates/management-controller/configmap.yaml`

Reply:
Setting `EXTERNAL_WEBHOOK_SERVER_BASE_URL` properly will be a little tricky.

You need to look at a variety of TLS and Ingress settings to properly determine if the base URL begins with http or https.

You can look at how we inject the API server's base URL into the controller for inspiration:

https://github.com/akuity/kargo/blob/4969e0c7f9e6734b4d0eafe7801bcbadb3243c35/charts/kargo/templates/controller/configmap.yaml#L12

https://github.com/akuity/kargo/blob/main/charts/kargo/templates/_helpers.tpl#L97-L106